### ISSUE TITLE
Snipe-it upgrade from v6.2.3 to v7.0.13 fails

### ISSUE THREAD

**Issue by HIS-David** [grokability/snipe-it#15790]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

We are using Snipe-It version 6.2.3 and are trying to upgrade to 7.0.13 but are having an issue with the database migration of "2023-12-14-032522_add_remote_ip_and_action_source_to_action_logs" due to max time execution exceeding 600 seconds.  We have around 5 million records in the action_logs table so will take a bit longer than 10 minutes.  

![image](https://github.com/user-attachments/assets/73d27b64-6751-4ae0-9fe5-e1cdaeb56bbd)

We have tried to update the php.ini with the following to see if we could pass this 600 seconds limit.
max_execution_time = 1200
max_execution_time = 0
memory_limit = 512M

### Reproduction steps

1. We have tried by installing a vanilla 7.0.13 and restoring the database and migrate but same issue occurred.  
2. Updated php.ini with max_execution_time = 1200, max_execution_time = 0, memory_limit = 512M to try to get past the 600 sec limit.
3. Check dependencies files.
4. Delete first year of action_log records to reduce overall size of table. This allowed the system to upgrade but had server 500 errors when updating any asset after migration had completed. 

### Expected behavior

attached update log and laravel log
[Laravel.txt](https://github.com/user-attachments/files/17681672/Laravel.txt)
[updatelog.txt](https://github.com/user-attachments/files/17681673/updatelog.txt)

### Snipe-IT Version

6.2.3

### Operating System

Windows

### Web Server

IIS

### PHP Version

8.1.28

### Operating System

Windows

### Browser

Edge

### Version

130.0.2849.68

### Device

n/a

### Operating System

n/a

### Browser

n/a

### Version

n/a

### Error messages

```shell
[2024-11-08 15:40:52] production.ERROR: Symfony\Component\ErrorHandler\Error\FatalError: Maximum execution time of 600 seconds exceeded in D:\snipe-it\vendor\laravel\framework\src\Illuminate\Database\Connection.php:587
Stack trace:
#0 {main}  
[2024-11-08 15:40:52] production.ERROR: Maximum execution time of 600 seconds exceeded {"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Maximum execution time of 600 seconds exceeded at D:\\snipe-it\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Connection.php:587)
[stacktrace]
#0 {main}
"}
```

**Comment by snipe** [comment]
I think the best route here (which is awful) is to manually add those columns and then insert the migration into the migrations table so it gets skipped on the next time around. It's odd that it would take too long when it's just adding the columns, not actually populating them with data - and 5 million isn't even really that much for MySQL.

Maybe @uberbrady has a better suggestion though.

**Comment by uberbrady** [comment]
This doesn't make sense - on most new-ish versions of MySQL, adding a nullable field as the last column is optimized to run very quickly. I suspect if you do the appropriate `ALTER TABLE blah ADD COLUMN` and see how long each of those statements take, you'll get a better idea. I imagine each one of those `ALTER` statements will be relatively slow, and adding up the time for all three will exceed the 10 minutes.

But I _don't_ understand why. Maybe really 'slow' storage?

Also the editing of the php.ini _should have_ helped - if you do `php -i | grep max_execution_time` it _should_ say your longer value. I'm not finding any places in Laravel itself where the max execution time is being adjusted, in grep'ing through the source.

So, in short - I don't know why this is happening; it shouldn't, on a few different levels. But @snipe 's workaround _should_ work.

**Comment by snipe** [comment]
Hi there - any updates?

**Comment by HIS-David** [comment]
> Hi there - any updates?

Planning on trying again this afternoon. Should have an update tomorrow morning.

**Comment by HIS-David** [comment]
I upgraded the server to PHP 8.3.14 and MariaDB 10.5.27.0 and changed the php.ini settings back to the default.  

![image](https://github.com/user-attachments/assets/af04428b-1db1-46a9-b8da-e9bb69171626)

This got us past the 2023_12-14_032522_add_remote_ip_and_action_source_to_action_logs migration but are now stuck on the following.

![image](https://github.com/user-attachments/assets/7957d5c6-83f8-46c3-9f26-459e82a0fd22)

Not sure how to re-run the migration as if i do "php artisan migrate" it just fails at the same section.

**Comment by snipe** [comment]
You can manually add that migration to the migrations table, and then re-run migrations.

**Comment by HIS-David** [comment]
I'm not sure how to do that.  Any pointers would be great. Thanks.

**Comment by HIS-David** [comment]
@snipe any update on if you could provide any pointer on how to "manually add that migration to the migrations table, and then re-run migrations."

Thanks.

**Comment by snipe** [comment]
Sure:

```
INSERT INTO migrations (migration, batch) VALUES ('2024_07_26_143301_add_checkout_for_all_types_to_accessories', 1);
```

then re-run migrations

**Comment by HIS-David** [comment]
@Snipe Thank you very much for the response.  That has allowed us to rerun the migration, and it completed.  I managed to log into the system but did have to move back to php 8.1.2.8 from 8.3.14 as I was getting a server 500 error with v7.0.13.  I'm not too concerned about this though.   I did try going to snipeit v7.1.15 using 'php upgrade.php' scripta and git but it just kept the system at v7.0.13. I can manually download the files and get there so not too much of a problem.  Thanks again.
